File: src/model/rules/inputRule/InputRuleImplBomb.java
package model.rules.inputRule;

import model.enums.BombPieceType;
import model.pieces.PieceImplBomb;
import model.structs.Move;
import model.structs.Point;

public class InputRuleImplBomb implements InputRule {

    private static final InputRuleImplBomb instance = new InputRuleImplBomb();

    public static InputRuleImplBomb getInputRule() {
        return instance;
    }

    @Override
    public Move ParseInput(String input) {
        boolean isBomb =
                input.charAt(0) == PieceImplBomb.CRATER_PIECE &&
                        !(input = input.substring(1)).isEmpty();
        if(input.length() != 2) {
            return null;
        }
        PieceImplBomb piece = new PieceImplBomb();
        piece.setType(
                isBomb ?
                BombPieceType.CRATER :
                BombPieceType.NORMAL);
        return new Move(
                new Point(),
                new Point(
                        InputRule.parseCol(input.charAt(1)),
                        InputRule.parseRow(input.charAt(0))),
                piece
        );
    }
}

File: src/model/rules/inputRule/InputRuleImplBomb.kt
package model.rules.inputRule

import model.enums.BombPieceType
import model.pieces.PieceImplBomb
import model.rules.inputRule.InputRule.Companion.parseCol
import model.rules.inputRule.InputRule.Companion.parseRow
import model.structs.Move
import model.structs.Point

class InputRuleImplBomb : InputRule {
    override fun parseInput(input: String): Move? {
        var input = input
        val isBomb =
            input[0] == PieceImplBomb.CRATER_PIECE &&
                    !(input.substring(1).also { input = it }).isEmpty()
        if (input.length != 2) {
            return null
        }
        val piece = PieceImplBomb()
        piece.type = if (isBomb) BombPieceType.CRATER else BombPieceType.NORMAL
        return Move(
            Point(),
            Point(
                parseCol(input[1]),
                parseRow(input[0])
            ),
            piece
        )
    }

    companion object {
        val inputRule: InputRuleImplBomb = InputRuleImplBomb()
    }
}
